content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -447,35 +449,95 @@ contexts:
+and **must** respond with a `TokenReview` object in the same version as the request.
 
-The POST body will be of the following format:
+{{< tabs name="TokenReview_request" >}}
+{{% tab name="authentication.k8s.io/v1" %}}
+{{< note >}}
+The Kubernetes API server defaults to sending `authentication.k8s.io/v1beta1` token reviews for backwards compatibility.
+To opt into receiving `authentication.k8s.io/v1` token reviews, the API server must be started with `--authentication-token-webhook-version=v1`.
+{{< /note >}}
 
-```json
+```yaml
+{
+  "apiVersion": "authentication.k8s.io/v1",
+  "kind": "TokenReview",
+  "spec": {
+    # opaque bearer token sent to the API server
+    "token": "014fbff9a07c...",
+   
+    # optional list of the identifiers the server presented with the token identifies as.
+    # Audience-aware token authenticators should verify the token was intended for at least one of the audiences in this list,
+    # and include the valid audiences for the token in the response status.
+    # If no audiences are provided, the token should be validated for purposes of authenticating to the Kubernetes API server.
+    "audiences": ["https://myserver.example.com", "https://myserver.internal.example.com"]
+  }
+}
+```
+{{% /tab %}}
+{{% tab name="authentication.k8s.io/v1beta1" %}}
+```yaml
 {
   "apiVersion": "authentication.k8s.io/v1beta1",
   "kind": "TokenReview",
   "spec": {
-    "token": "(BEARERTOKEN)"
+    # opaque bearer token sent to the API server
+    "token": "014fbff9a07c...",
+   
+    # optional list of the identifiers the server presented with the token identifies as.
+    # Audience-aware token authenticators should verify the token was intended for at least one of the audiences in this list,
+    # and include the valid audiences for the token in the response status.
+    # If no audiences are provided, the token should be validated for purposes of authenticating to the Kubernetes API server.
+    "audiences": ["https://myserver.example.com", "https://myserver.internal.example.com"]
   }
 }
 ```
+{{% /tab %}}
+{{< /tabs >}}
 
-The remote service is expected to fill the `status` field of
-the request to indicate the success of the login. The response body's `spec`
-field is ignored and may be omitted. A successful validation of the bearer
-token would return:
+The remote service is expected to fill the `status` field of the request to indicate the success of the login.
+The response body's `spec` field is ignored and may be omitted.
+The remote service must return a response using the same `TokenReview` API version it was sent.
+A successful validation of the bearer token would return:
 
-```json
+{{< tabs name="TokenReview_response_success" >}}
+{{% tab name="authentication.k8s.io/v1" %}}
+```yaml
+{
+  "apiVersion": "authentication.k8s.io/v1",
+  "kind": "TokenReview",
+  "status": {
+    "authenticated": true,
+    "user": {
+      "username": "janedoe@example.com",
```

Comment:
Related to the query about `uid`: is username mandatory? If a success response omits `username`, is that valid? If not, is `""` valid-but-daft?

Reply:
updated with "required"... I think an empty string might actually be tolerated currently, but I'd consider that a bug we should fix to prevent typos/accidents rather than a feature we should preserve